### PR TITLE
fix: rotating SVG should not cause parent overflow

### LINKED ISSUE #30610: [Bug]: Vertical scrollbar appears occasionally when Spinner is the only element in an `overflow: auto` container
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (20) x64 Intel(R) Core(TM) i9-10900X CPU @ 3.70GHz
    Memory: 20.60 GB / 31.21 GB
    Container: Yes
    Shell: 5.8.1 - /usr/bin/zsh
```

Browser: Edge 121.0.2277.128
Reduce motion: false
window.devicePixelRatio: 1.25

### Are you reporting Accessibility issue?

no

### Reproduction

https://codesandbox.io/p/sandbox/adoring-feather-mcj59v

## Actual Behavior
The container sometimes overflow. The vertical scrollbar appears occasionally.

## Expected Behavior
The container will not overflow.

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

SharePoint

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by smhigley**
## Previous Behavior

The rotation of the SVG sometimes triggered parent scroll overflow (i.e. when rotated 45deg, the diagonal dimension is longer than the 0deg height).

## New Behavior

Adds an `overflow: hidden` to the span wrapping the SVG

## Related Issue(s)

- Fixes #30610

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/Avatar-mount_base.html">631</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/Avatar-mount.html">646</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/Button-mount_base.html">290</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/Button-mount.html">309</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/Field-mount_base.html">1118</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/Field-mount.html">1106</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/FluentProvider-mount_base.html">693</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/FluentProvider-mount.html">721</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/FluentProviderWithTheme-mount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/FluentProviderWithTheme-mount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">61</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">63</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">78</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">80</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/MakeStyles-mount_base.html">859</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/MakeStyles-mount.html">868</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/Persona-mount_base.html">1744</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/Persona-mount.html">1712</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/SpinButton-mount_base.html">1371</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30615/perf-test-react-components/SpinButton-mount.html">1382</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr> | `23.517 kB`<br />`8.512 kB` | `23.619 kB`<br />`8.557 kB` | `102 B` <br />`45 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `70.732 kB`<br />`20.465 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `217.207 kB`<br />`61.943 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `43.219 kB`<br />`14.299 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `7.944 kB`<br />`2.588 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/2e8b33675468ee9bbf88bb14dcca89a0173fb4a3'>2e8b33675468ee9bbf88bb14dcca89a0173fb4a3</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-spinner-503d4d53-0c62-4c35-b641-f805615cd2c8.json
Comment (fabricteam) [inline change/@fluentui-react-spinner-503d4d53-0c62-4c35-b641-f805615cd2c8.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30615&commitId=73417a652184f1cec402c6500d5f9d1470c5c758&env=prod&clientType=FLUENTUI)


  <details><summary><b>Spinner converged</b> 12 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Spinner converged.Inverted with Label - High Contrast.chromium.png|5| Changed|
|Spinner converged.Primary - Dark Mode.chromium.png|1| Changed|
|Spinner converged.Primary with Label.chromium.png|3| Changed|
|Spinner converged.Inverted with Label.chromium.png|3| Changed|
|Spinner converged.Primary with Label - High Contrast.chromium.png|5| Changed|
|Spinner converged.Primary with Label - RTL.chromium.png|4| Changed|
|Spinner converged.Animation.chromium.png|9| Changed|
|Spinner converged.Inverted with Label - Dark Mode.chromium.png|3| Changed|
|Spinner converged.Primary - High Contrast.chromium.png|1| Changed|
|Spinner converged.Primary + size.chromium.png|6| Changed|
|Spinner converged.Primary with Label - Dark Mode.chromium.png|2| Changed|
|Spinner converged.Inverted - High Contrast.chromium.png|1| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Comment by tomi-msft** [comment]
This PR might be affected by the changes in the `Spinner` animation done by this one: #30567. We should wait until that one gets merged and see what the differences are in behavior. What do you think @smhigley

**Comment by smhigley** [comment]
@tomi-msft In theory I don't think this will impact the other PR since the overflow is going on the parent and the root problem will remain (since we're keeping rotation styles).

I'm happy to have this wait just in case though!

**Review by tomi-msft — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**
